docs/pages/CodeAnalysis.md

```diff
@@ -0,0 +1,227 @@
+# Code Analysis
+
+During the build process, UI5 Tooling executes a static code analysis of your project. In the following sections the term "simple literal" is used for the following language features:
+
+- string literals
+- numeric literals
+- Boolean literals
+- `null` / `undefined`
+- template literals without any expressions
+- arrays of the previous (spread operators are ignored)
+- object literals with "simple literals" used as keys and values (again, spread operators are ignored)
+
+## Dependency Analysis
+
+UI5 Tooling extracts dependency information from a project's code as outlined in the following sections.
+The APIs described there require the usage of "simple literals" when declaring dependencies.
+
+### JSModule Analyzer
+
+The following APIs are analyzed by the JSModule Analyzer:
+
+- sap.ui.define
+- sap.ui.require
+- jQuery.sap.declare (deprecated)
+- jQuery.sap.require (deprecated)
+- sap.ui.requireSync (deprecated)
+- sap.ui.preload (restricted)
+- sap.ui.require.preload (restricted)
+
+The [JSModule Analyzer](https://github.com/SAP/ui5-builder/blob/main/lib/lbt/analyzer/JSModuleAnalyzer.js) uses the ["Abstract Syntax Tree"](https://en.wikipedia.org/wiki/Abstract_syntax_tree) (AST) of a JavaScript file to decide whether a code block is executed *conditionally* or *unconditionally*.
+
+The analyzer uses a set of rules to decide whether one of the above APIs is called whenever the module is executed or whether the API is only called under certain conditions.
+
+For example, top-level code is always executed. Flow-control statements in JavaScript imply that certain blocks of code are only executed under certain conditions (for example, `if` blocks, `else` blocks, ...). Besides these inherent JavaScript rules, further common patterns are known to the analyzer, e.g. immediately invoked function expressions or the factor function of AMD modules.
+
+Any dependencies found that - according to these rules - are always executed, are collected as eager (or standard) dependencies. Dependencies that are found on a code path that depends on certain conditions are collected as conditional dependencies.
+
+The bundling implemented by UI5 Tooling can either follow only eager dependencies (resolve:`true`) or additionally conditional dependencies (resolveConditional). For more information, see [Custom Bundling](https://sap.github.io/ui5-tooling/v3/pages/Configuration/#custom-bundling).
```

Comment:
sholdn't config properties (resolve, resolveConditional) be wrapped in backticks?

Reply:
done